game-core/src/main/java/games/strategy/util/PointFileReaderWriter.java

```diff
@@ -256,115 +199,62 @@ public static void writeOneToManyPlacements(final OutputStream sink,
-            }
-            if (current == '>') {
-              // end poly
-              createPolygonFromPoints(polygons, points);
-              points.clear();
-              // break from while(true)
-              break;
-            }
-          }
-        }
-        index++;
-      }
-      if (mapping.containsKey(name)) {
-        throw new IOException("name found twice:" + name);
+      final Matcher pointMatcher = pointPattern.matcher(polyMatcher.group());
+      while (pointMatcher.find()) {
+        points.add(new Point(Integer.parseInt(pointMatcher.group(1)), Integer.parseInt(pointMatcher.group(2))));
       }
-      mapping.put(name, polygons);
-    } catch (final StringIndexOutOfBoundsException e) {
-      throw new IllegalStateException("Invalid line:" + line, e);
+      polygons.add(createPolygonFromPoints(points));
     }
+    mapping.put(name, polygons);
   }
 
-  private static void createPolygonFromPoints(final Collection<Polygon> polygons, final List<Point> points) {
+  private static Polygon createPolygonFromPoints(final List<Point> points) {
     final int[] pointsX = new int[points.size()];
     final int[] pointsY = new int[points.size()];
     for (int i = 0; i < points.size(); i++) {
       final Point p = points.get(i);
       pointsX[i] = p.x;
       pointsY[i] = p.y;
     }
-    polygons.add(new Polygon(pointsX, pointsY, pointsX.length));
+    return new Polygon(pointsX, pointsY, pointsX.length);
   }
 
-  private static Tuple<String, List<Point>> readMultiple(final String line, final Map<String, List<Point>> mapping)
-      throws IOException {
-    final StringTokenizer tokens = new StringTokenizer(line, "");
-    final String name = tokens.nextToken("(").trim();
+  private static Tuple<String, List<Point>> readMultiple(final String line, final Map<String, List<Point>> mapping) {
+    final String name = line.substring(0, line.indexOf("(")).trim();
     if (mapping.containsKey(name)) {
-      throw new IOException("name found twice:" + name);
+      throw new IllegalArgumentException("name found twice:" + name);
     }
+    final Matcher matcher = pointPattern.matcher(line);
     final List<Point> points = new ArrayList<>();
-    while (tokens.hasMoreTokens()) {
-      final String stringX = tokens.nextToken(",(), ");
-      if (!tokens.hasMoreTokens()) {
-        continue;
-      }
-      final String stringY = tokens.nextToken(",() ");
-      final int x = Integer.parseInt(stringX);
-      final int y = Integer.parseInt(stringY);
-      points.add(new Point(x, y));
+    while (matcher.find()) {
+      points.add(new Point(Integer.parseInt(matcher.group(1)), Integer.parseInt(matcher.group(2))));
     }
     mapping.put(name, points);
     return Tuple.of(name, points);
   }
+
+  private static void readStream(final InputStream stream, Consumer<String> lineParser)
+      throws IOException {
+    try (Reader inputStreamReader = new InputStreamReader(new CloseShieldInputStream(stream), StandardCharsets.UTF_8);
```

Comment:
Does `new CloseShieldInputStream(stream)` need to be closed?

Reply:
`new CloseShieldInputStream(stream)` is a wrapper that prevents the underlying stream to be closed, even if close is being called.
So no.